### DISCUSSION TITLE
What are you guys up to?

### DISCUSSION THREAD

**Discussion by metarama** [aws-samples/aws-saas-factory-ref-solution-serverless-saas#44] [Ideas]
Thanks to the team that maintains this ref app template.

I see that this ref app template is not evolving and has been stagnant for a while. Is there nothing else that can be improved further in the nerveless ref app template?

Here are some suggestions if you are looking for ideas and some interesting dev evangelism ideas:

1. Show how the server side can be built in Typescript as well as Python. (Others have asked for it)
2. The SAM template.yml is hard to maintain. Show how  this can be done with 100% modular CDK in typescript and python.
3. Show DynamoDB single table design pattern. This is a ref app and that best practice must be demonstrated here.
4. Most of us are transitioning from SQL to DynamoDB. Show how the MicroService implementation separates DynamoDB code, the data model, and business logic separate. I see Alex Debrie has a decent starting point for this separation as demonstrated in the E-Commerce store implementation YouTube video.

I have more if you are interested.

Regards,
Rama

**Comment by anubhavynr** [comment]
Hi Rama,

Thanks a lot for the feedback. We have been making changes as needed. As an example, we recently added new Lab7 for cost per tenant in the accompanying workshop that comes along with this reference architecture. We also made some changes with Cognito amplify library recently. We have plans to enhance the DevOps footprint in future.

1) Idea is to demonstrate the SaaS general best practices. I understand that people might have their choice of programming languages.
2) I agree, we are working on migrating to complete CDK in future. Currently the DevOps pipeline is being maintained in CDK. So we have both SAM and CDK working in tandem currently.
3) We already have the DynamoDB single table design pattern. If you onboard a customer as a platinum tier it creates table per tenant.
4) When you design your application, we encourage you to take concepts as described here and combine with other best practices as you mentioned above. Given the open source nature of the project, we encourage customers to submit pull requests. I will happy to review and merge.

Please let me know if you have more feedback.

**Comment by metarama** [comment]
Thank you Anubhav. I look forward to the DevOps enhancements to this ref app. What is the ETA for this work to hit the repo?

This ref app does a lot of heavy lifting towards developing a real world multi-tenant SaaS app. In your opinion and experience what are the additional elements (which are not handled in this ref app) that need to be addressed in a production grade SaaS app? Could you please list these missing additional elements?

As an example,  you mentioned (in another discussion) separation of Shared services into a separate repository with its own CI/CD pipeline.

For item 3 above by 'single table design' I meant the idea championed by [Debrie](https://www.alexdebrie.com/posts/dynamodb-single-table/). The platinum silo is also super cool.
